aptos-move/move-examples/token_objects/ambassador/move/sources/ambassador.move

```diff
@@ -47,12 +49,16 @@ module token_objects::ambassador {
 
     /// The ambassador token
     struct AmbassadorToken has key {
+        /// Used to mutate the token uri
+        mutator_ref: token::MutatorRef,
         /// Used to burn.
         burn_ref: token::BurnRef,
         /// Used to mutate properties
         property_mutator_ref: property_map::MutatorRef,
         /// Used to emit LevelUpdateEvent
         level_update_events: event::EventHandle<LevelUpdateEvent>,
+        /// the base URI of the token
+        base_uri: String,
```

Comment:
This is an interesting choice to have this here vs the collection since this does enable more parallelism for operations that modify the rank of the token. I'd add a line here (or in the tutorial to be written) to explain this

Reply:
"parallelism" is an interesting point! `base_uri` here is intended to be the base URI for a token rather than a collection. In this example, each token has three URIs for gold, silver, and bronze images, so needs a common prefix.